### ISSUE TITLE
[NavigationMenu] Open state functionality

### ISSUE THREAD

**Issue by haniotis** [radix-ui/primitives#1301]
### Overview

Add `open` state to `NavigationMenu.Root` so we can override opening/closing states for use with external lib (e.g. Framer Motion).

Same functionality as found in `Collapsible`, `DropdownMenu`, `Dialog`, etc.

Edit: There's also no way of closing the menu when navigation between pages in a Next.js app and the menu stays open (located in global layout outside page content)

Thank you!

**Comment by andy-hook** [comment]
Hi @weskor , just to clarify, is your issue that the menu isn't closing when navigating client side to another route in your app? i.e. the `open` state of the menu is persisting on route change?

**Comment by weskor** [comment]
Hi @andy-hook, yes that seems to be the case. I could not find anything in the documentation regarding closing. Not sure if this the intended behavior and that I have to implement something for this.

**Comment by andy-hook** [comment]
> I could not find anything in the documentation regarding closing. Not sure if this the intended behavior and that I have to implement something for this.

@weskor This should be being handled for you out of the box to be honest so seems like an oversight on our part.

We'll raise a fix soon but in the meantime you can control the value yourself and set it to an empty string in order to close the menu manually

https://codesandbox.io/s/distracted-gwen-domugj?file=/pages/_app.js

Hope this helps 🙏

**Comment by weskor** [comment]
Hi @andy-hook, thank you so much 🙏 Really love what Radix is doing and thanks for the quick reaction.

**Comment by andy-hook** [comment]
> Add open state to NavigationMenu.Root so we can override opening/closing states for use with external lib 

@haniotis Just to clarify on this point, the functionality for managing the "open" state does exist but rather than a `boolean` value per `Collapsible`, `DropdownMenu`, `Dialog` this menu uses a `value` prop which controls the currently active `Item`. Out of the box we generate these but if you need to control the currently active item you can assign your own to each item.

https://www.radix-ui.com/docs/primitives/components/navigation-menu#item

Happy to provide an example of animating this with framer if you're still having issues.

**Comment by haniotis** [comment]
@andy-hook Yup! That's what I ended up figuring out therefore closed the issue.

Wouldn't mind an example with framer-motion though - that would be greatly appreciated

**Comment by samburgers** [comment]
thanks for this! i implemented a slight change to @andy-hook example, hooking into route changes rather than click events - meaning it should catch things like browser fwd/back buttons and keyboard shortcuts etc too

```
  const [value, setValue] = React.useState("");
  const { asPath } = useRouter();

  React.useEffect(() => {
    setValue("");
  }, [asPath]);
  ```
https://codesandbox.io/s/mystifying-germain-mlb900?file=/pages/_app.js

**Comment by andy-hook** [comment]
> so the recommended way is to watch for a router change and then set the component to close

Yes

**Comment by andy-hook** [comment]
@shawnmclean can you provide a codesandbox showing your existing implementation? would help us provide some guidance.

**Comment by maurer2** [comment]
Hello, I believe it might be possible to avoid `useEffect` by using the pathname that is returned from the `usePathname` hook as a key for the navigation component. This would mean that the Navigation component would be reset whenever the pathname changes. The approach is described here: https://react.dev/learn/you-might-not-need-an-effect#resetting-all-state-when-a-prop-changes

It could look something like this:

NavigationWrapper component

```tsx
'use client'

import { usePathname } from 'next/navigation';
import Navigation from './navigation';

export function NavigationWrapper() {
  const pathname = usePathname();

  return (
    <Navigation key={pathname} />
  )
}
```

Cheers

**Comment by andy-hook** [comment]
> I believe it might be possible to avoid useEffect here by using the pathname that is returned from the usePathname hook as a key for the navigation component.

Also a good option providing you don't have additional state that needs to be persisted.

**Comment by DevPro15** [comment]
@andy-hook Thanks, was able to fix it
